File: 2483-minimum-penalty-for-a-shop/2483-minimum-penalty-for-a-shop.js
/**
 * @param {string} customers
 * @return {number}
 */
var bestClosingTime = function(customers) {
    let pen = 0;
    const n = customers.length;
    for (let i = 0; i < n; i++) {
        if (customers.charCodeAt(i) === 89) pen++; 
    }

    let best = pen;
    let ans = 0;

    for (let i = 0; i < n; i++) {
        pen += (customers.charCodeAt(i) === 89) ? -1 : 1;

        if (pen < best) {
            best = pen;
            ans = i + 1;
        }
    }
    return ans;
};

File: 2483-minimum-penalty-for-a-shop/2483-minimum-penalty-for-a-shop.cs
public class Solution {
    public int BestClosingTime(string customers) {
        int pen = 0;
        foreach (char c in customers)
            if (c == 'Y') pen++;

        int best = pen, ans = 0;

        for (int i = 0; i < customers.Length; i++) {
            if (customers[i] == 'Y')
                pen--;  
            else
                pen++;  

            if (pen < best) {
                best = pen;
                ans = i + 1;
            }
        }
        return ans;
    }
}
